auth/oras/client_test.go

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package oras
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	"oras.land/oras-go/v2/registry/remote/auth"
+)
+
+func TestAuthClient(t *testing.T) {
+	testServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if r.Method != http.MethodGet {
+			t.Errorf("unexpected access: %s %s", r.Method, r.URL)
+			w.WriteHeader(http.StatusMethodNotAllowed)
+			return
+		}
+		// writes uuid of the request in the response header
+		uuid := r.Header.Get("x-ms-correlation-request-id")
+		if uuid != "" {
+			w.Header().Set("x-ms-correlation-request-id", uuid)
+		}
+	}))
+	defer testServer.Close()
+
+	uri, err := url.Parse(testServer.URL)
+	if err != nil {
+		t.Fatalf("invalid test http server: %v", err)
+	}
+
+	testClient := NewClient(ClientOptions{
+		Credential: Credential("testUser", "testPassword"),
+	})
+	req := http.Request{URL: uri, Header: http.Header{}}
```

Comment:
It is not common to create `http.Request` like this. Try `http.NewRequest()`.

Reply:
changed